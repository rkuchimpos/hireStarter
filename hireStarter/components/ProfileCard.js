import React from "react";
import {
  Dimensions,
  Image,
  Platform,
  ScrollView,
  StyleSheet,
  Text,
  View
} from "react-native";
import SwiperFlatList from "react-native-swiper-flatlist";
import Skill from "./Skill";

const { width } = Dimensions.get("window");

class ProfileCard extends React.Component {
  render() {
    return (
      <View style={styles.cardWrapper}>
        <View style={styles.card}>
          <View>
            <SwiperFlatList
              index={0}
              data={this.props.photos}
              renderItem={({ item }) => (
                <Image
                  resizeMode="cover"
                  style={styles.image}
                  source={{ uri: item }}
                />
              )}
              showPagination
            />
          </View>
          <ScrollView>
          <View style={styles.profileInfo}>
            <Text style={styles.name}>{this.props.name}</Text>
            <Text style={styles.subtitle}>{this.props.organization}</Text>
            <Text style={styles.subtitle}>{this.props.city}</Text>
            <Text style={styles.description}>{this.props.description}</Text>
            <View style={styles.skillList}>
              {this.props.skills.map(skill => (
                <View style={{ marginRight: 5, marginBottom: 5 }} key={skill}>
                  <Skill skill={skill} />
                </View>
              ))}
            </View>
          </View>
          </ScrollView>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  cardWrapper: {
    ...Platform.select({
      ios: {
        shadowColor: "#000",
        shadowRadius: 1,
        shadowOpacity: 0.5,
        shadowOffset: {
          width: 0,
          height: 0
        }
      }
    })
  },
  card: {
    overflow: "hidden",
    backgroundColor: "#fff",
    borderRadius: 10,
    ...Platform.select({
      android: {
        elevation: 1
      }
    })
  },
  image: {
    width: width - 20,
    height: 350
  },
  profileInfo: {
    margin: 10,
    //height: 220,
  },
  name: {
    //fontWeight: "bold",
    fontSize: 24,
    color: "#606060"
  },
  subtitle: {
    color: "#909090",
    fontSize: 14
  },
  description: {
    color: "#404040",
    fontSize: 14,
    marginTop: 15,
  },
  skillList: {
    marginTop: 20,
    flexDirection: "row",
    flexWrap: "wrap",
    justifyContent: "center"
  }
});

export default ProfileCard;
