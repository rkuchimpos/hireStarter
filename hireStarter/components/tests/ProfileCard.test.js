import React from "react";
import renderer from "react-test-renderer";
import ProfileCard from "../ProfileCard";

it("ProfileCard renders correctly", () => {
  const tree = renderer
    .create(
      <ProfileCard
        name={"Joe Bruin"}
        location={"Los Angeles"}
        description={"I am a software engineer"}
        skills={["Java", "C++"]}
        photos={[
          "https://i.imgur.com/cMFc42W.png",
          "https://i.imgur.com/6B55OIA.png"
        ]}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
