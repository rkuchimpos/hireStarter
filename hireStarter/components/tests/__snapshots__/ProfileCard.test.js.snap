// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileCard renders correctly 1`] = `
<View
  style={
    Object {
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 10,
        "overflow": "hidden",
      }
    }
  >
    <View>
      <RCTScrollView
        data={
          Array [
            "https://i.imgur.com/cMFc42W.png",
            "https://i.imgur.com/6B55OIA.png",
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        initialNumToRender={1}
        initialScrollIndex={0}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        onScrollToIndexFailed={[Function]}
        onViewableItemsChanged={[Function]}
        pagingEnabled={true}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEnabled={true}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfig={
          Object {
            "itemVisiblePercentThreshold": 60,
            "minimumViewTime": 200,
          }
        }
        viewabilityConfigCallbackPairs={
          Array [
            Object {
              "onViewableItemsChanged": [Function],
              "viewabilityConfig": Object {
                "itemVisiblePercentThreshold": 60,
                "minimumViewTime": 200,
              },
            },
          ]
        }
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <Image
              resizeMode="cover"
              source={
                Object {
                  "uri": "https://i.imgur.com/cMFc42W.png",
                }
              }
              style={
                Object {
                  "height": 350,
                  "width": 730,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "width": 0,
              }
            }
          />
        </View>
      </RCTScrollView>
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginVertical": 16.675,
              "position": "absolute",
              "width": "100%",
            },
            Object {},
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 25,
              "height": 28.125,
              "marginHorizontal": 18.75,
              "opacity": 1,
              "width": 28.125,
            }
          }
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "gray",
              "borderRadius": 25,
              "height": 28.125,
              "marginHorizontal": 18.75,
              "opacity": 1,
              "width": 28.125,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "margin": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#606060",
            "fontSize": 24,
          }
        }
      >
        Joe Bruin
      </Text>
      <Text
        style={
          Object {
            "color": "#909090",
            "fontSize": 14,
          }
        }
      >
        Los Angeles
      </Text>
      <Text
        style={
          Object {
            "color": "#404040",
            "fontSize": 14,
            "marginTop": 20,
          }
        }
      >
        I am a software engineer
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "center",
            "marginTop": 20,
          }
        }
      >
        <View
          style={
            Object {
              "marginBottom": 5,
              "marginRight": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#4A00E0",
                "borderRadius": 20,
                "padding": 8,
                "paddingLeft": 10,
                "paddingRight": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 14,
                }
              }
            >
              Java
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 5,
              "marginRight": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#4A00E0",
                "borderRadius": 20,
                "padding": 8,
                "paddingLeft": 10,
                "paddingRight": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 14,
                }
              }
            >
              C++
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
